pkg/sql/truncate.go

```diff
@@ -215,9 +215,15 @@ func (p *planner) truncateTable(ctx context.Context, id descpb.ID, jobDesc strin
 	}
 
 	// Create new ID's for all of the indexes in the table.
-	version := p.ExecCfg().Settings.Version.ActiveVersion(ctx)
-	if err := tableDesc.AllocateIDs(ctx, version); err != nil {
-		return err
+	{
+		version := p.ExecCfg().Settings.Version.ActiveVersion(ctx)
+		// Temporarily empty the mutation jobs slice for AllocateIDs to not fail.
+		mutationJobs := tableDesc.MutationJobs
+		tableDesc.MutationJobs = nil
+		if err := tableDesc.AllocateIDs(ctx, version); err != nil {
+			return err
+		}
+		tableDesc.MutationJobs = mutationJobs
```

Comment:
What's the reason for this change in `sql/truncate.go`? Why isn't it previously a problem?

Reply:
Good question. AllocateIDs, which is a bad function but which nobody wants to touch, validates the descriptor. Prior to this call, the Mutations slice was zeroed, but the MutationJobs slice only gets zeroed later on after doing things to the jobs. 

So, this is a bit dirty, but hey, reimplementing TRUNCATE in the declarative schema changer should be easy, so I don't feel bad about it.